sql/core/src/main/scala/org/apache/spark/sql/execution/columnar/InMemoryTableScanExec.scala

```diff
@@ -65,6 +65,11 @@ private[sql] case class InMemoryTableScanExec(
     case EqualTo(l: Literal, a: AttributeReference) =>
       statsFor(a).lowerBound <= l && l <= statsFor(a).upperBound
 
+    case EqualNullSafe(a: AttributeReference, l: Literal) if !l.nullable =>
+      statsFor(a).lowerBound <= l && l <= statsFor(a).upperBound
+    case EqualNullSafe(l: Literal, a: AttributeReference) if !l.nullable =>
```

Comment:
Actually, it seems this condition, `!l.nullable` is not needed since `field <=> NULL` is converted to `IsNull(field)` and it is always `Literal` but I just added this just in case.